### ISSUE TITLE
CTRL+Space causes Heidi to hang for several minutes while loading db schema

### ISSUE THREAD

**Issue by lieszkol** [HeidiSQL/HeidiSQL#1869]
**Is your feature request related to a problem? Please describe.**
After Heidi connects to a database in the editor window if I accidentally type CTRL + Space the application hangs while trying to load the database schema. This is particularly annoying on databases with lots of stored procedures and other logic. This has annoyed me for years now.

**Describe the solution you'd like**
Please provide an option NOT to download the stored procedure schemas from the database for the autocomplete cache. The operation takes so long because it executes "SHOW CREATE PROCEDURE…" for all of the procedures (and functions etc.) in the database, but I have no need for autocompletion or or hints for stored procedure parameters. 

**Describe alternatives you've considered**
It's possible to turn off autocompletion completely.

**Additional context**
![image](https://github.com/HeidiSQL/HeidiSQL/assets/16566226/247442e2-7953-4712-a956-6f5ead3a1b61)

**Comment by ansgarbecker** [comment]
Yes, this is indeed a problem. These `SHOW CREATE ...` queries are only fired for getting the procedure arguments into the proposal:

![grafik](https://github.com/HeidiSQL/HeidiSQL/assets/7986591/f27c5364-3184-4b44-a801-c4c68d429a3a)

I can prevent that query if the number of procedures is larger than 10 (?). Or probably only fire it for the first 10 procedures.

**Comment by lieszkol** [comment]
Preventing the query if the number of procedures is larger than 10 would certainly work for me! Or adding an option to the "Auto completion proposal" section of the SQL Perferences page. In any case, thank you for all your work!!

**Comment by ansgarbecker** [comment]
Note the option to disable the completion proposal is already there:

![grafik](https://github.com/HeidiSQL/HeidiSQL/assets/7986591/834b31bb-b80f-4893-b608-95e95904248b)

I'm thinking a feature to limit that query to 10 procedures is still a good idea.

**Comment by lieszkol** [comment]
Sorry, I meant a possibility would be to add another option such as "Enable procedure argument hints" or similar which controls whether these are queried from the db.
